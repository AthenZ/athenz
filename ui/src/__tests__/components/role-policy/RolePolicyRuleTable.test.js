/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import RolePolicyRuleTable from '../../../components/role-policy/RolePolicyRuleTable';
import {
    buildPoliciesForState,
    getStateWithPolicies,
    renderWithRedux,
} from '../../../tests_utils/ComponentsTestUtils';

describe('PolicyRuleTable', () => {
    it('should render', () => {
        const domain = 'home.pgote';
        const policyName = 'test';
        const policies = buildPoliciesForState(
            {
                [`${domain}:policy.${policyName}:0`]: {
                    name: `${domain}:policy.${policyName}`,
                    modified: '2022-07-13T09:48:22.510Z',
                    version: '0',
                    active: true,
                    assertions: {
                        11921: {
                            role: 'home.pgote:role.allunixusers',
                            resource: 'home.pgote:aaaa',
                            action: 'aaa',
                            effect: 'ALLOW',
                            id: 11921,
                        },
                    },
                },
            },
            domain
        );

        const { getByTestId } = renderWithRedux(
            <table>
                <tbody>
                    <tr>
                        <RolePolicyRuleTable
                            showTable={true}
                            color={'white'}
                            number={0}
                            domain={domain}
                            name={policyName}
                        />
                    </tr>
                </tbody>
            </table>,
            getStateWithPolicies(policies)
        );
        const ruleTable = getByTestId('role-policy-rule-table');
        expect(ruleTable).toMatchSnapshot();
    });

    it('should render empty', () => {
        const { getByTestId } = renderWithRedux(
            <table>
                <tbody>
                    <tr>
                        <RolePolicyRuleTable
                            showTable={false}
                            color={'white'}
                            number={0}
                        />
                    </tr>
                </tbody>
            </table>
        );
        const ruleTable = null;
        expect(ruleTable).toMatchSnapshot();
    });
});
