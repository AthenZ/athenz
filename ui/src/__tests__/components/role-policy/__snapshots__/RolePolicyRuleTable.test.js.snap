// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyRuleTable should render 1`] = `
.emotion-0 {
  width: 100%;
  border-spacing: 0;
  display: table;
  border-collapse: separate;
}

.emotion-2 {
  text-align: left;
  border-bottom: 2px solid #d5d5d5;
  color: #303030;
  font-weight: 600;
  font-size: 16px;
  padding-bottom: 2px;
  vertical-align: top;
  padding: 10px 10px 0px 15px;
  word-break: break-all;
}

.emotion-4 {
  text-align: right;
  border-bottom: 2px solid #d5d5d5;
  color: #3570F4;
  font-weight: 300;
  font-size: 14px;
  padding-bottom: 2px;
  vertical-align: top;
  padding: 10px 10px 0px 15px;
  word-break: break-all;
}

.emotion-6 {
  color: #3570f4;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-8 {
  margin: 15px 0 0 0;
}

.emotion-12 {
  border-bottom: 2px solid #d5d5d5;
  color: #9a9a9a;
  font-weight: 600;
  font-size: 1.2rem;
  padding-bottom: 5px;
  vertical-align: top;
  text-transform: uppercase;
  padding: 5px 0 5px 15px;
  text-align: left;
  background-color: white;
}

.emotion-20 {
  border-bottom: 2px solid #d5d5d5;
  color: #9a9a9a;
  font-weight: 600;
  padding-bottom: 5px;
  vertical-align: top;
  text-transform: uppercase;
  width: 120px;
  text-align: center;
  background-color: white;
}

<table
  class="emotion-0 emotion-1"
  data-testid="role-policy-rule-table"
>
  <tbody>
    <tr>
      <th
        class="emotion-2 emotion-3"
        color="#303030"
      >
        Rule Details(
        1
        )
      </th>
      <th
        class="emotion-4 emotion-3"
        color="#3570F4"
      >
        <a
          class="emotion-6 emotion-7"
        >
          Add rule
        </a>
      </th>
    </tr>
    <tr>
      <td
        colspan="4"
      />
    </tr>
    <tr>
      <td
        colspan="4"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <table
            class="emotion-0 emotion-1"
          >
            <thead>
              <tr>
                <th
                  class="emotion-12 emotion-13"
                >
                  Effect
                </th>
                <th
                  class="emotion-12 emotion-13"
                >
                  Action
                </th>
                <th
                  class="emotion-12 emotion-13"
                >
                  Role
                </th>
                <th
                  class="emotion-12 emotion-13"
                >
                  Resource
                </th>
                <th
                  class="emotion-20 emotion-21"
                />
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`PolicyRuleTable should render empty 1`] = `null`;
